frontend/packages/client/src/api/notificationService.js

```diff
@@ -13,24 +14,40 @@ const options = {
   headers: { accept: 'application/json' },
 };
 
-export const startLeanplum = () => {
+const getDesiredAttributes = (communitySubIntentions) => {
+  //subscribeUpdateIntentions = [{communityId:"1", subscribeIntention:"subscribe"},{communityId:"2",subscribeIntention:"unsubscribe"}]
+  return communitySubIntentions
+    .map(({ communityId, subscribeIntention }) => ({
+      key: `community${communityId}`,
+      value:
+        subscribeIntention === subscribeNotificationIntentions.subscribe
+          ? 'True'
+          : 'False',
+    }))
+    .reduce((acc, curr) => {
+      const { key, value } = curr;
+      acc[key] = value;
+      return acc;
+    }, {});
+};
+export const startLeanplumForUser = (walletId) => {
   const IS_LOCAL_DEV = process.env.REACT_APP_APP_ENV === 'development';
 
   if (IS_LOCAL_DEV) {
     Leanplum.setAppIdForDevelopmentMode(LEANPLUM_APP_ID, LEANPLUM_DEV_KEY);
   } else {
     Leanplum.setAppIdForProductionMode(LEANPLUM_APP_ID, LEANPLUM_PROD_KEY);
   }
-
-  Leanplum.start();
-};
-
-export const setUserId = async (walletId) => {
-  try {
-    Leanplum.setUserId(walletId);
-  } catch (e) {
-    throw new Error(e);
-  }
+  return new Promise((resolve, reject) => {
```

Comment:
Returning a promise is necessary for ensuring the sequence of execution, we want start leanplum first then get user settings